### PR TITLE
Feature/1126 costs

### PR DESCRIPTION
**Pull request by DougC**
I believe this is functionally working against the acceptance criteria. I've also implemented the "delete" link, which is actually covered under 1337, but it would be hard to test without it.

The layout is broken. For some reason the right column appears below the form. @jon-shipley can you have a look at this? I really can't see how it differs from sectionForm.scala.html. And the table needs some styling.

### CODE REVIEW

**Comment by DougC** [comment]
I've merged in the changes from master. Had some conflicts on the costItemForm but I think I got it merged okay.

**Comment by DougC** [comment]
Looking for 2 approvals on this, I think. The code definitely needs some tidying up

**Comment by jon-shipley** [comment]
Sidebar rendering tweaked - it's now back in place.

**Comment by DougC** [comment]
Right. I think this is ready to merge (along with the corresponding `rifs-business` PR. Looking for two approvals for this.

File: src/main/scala/controllers/ActionHandler.scala [hunk 1]
```diff
@@ -5,35 +5,54 @@ import javax.inject.Inject
...
-      case true => applications.deleteSection(id, sectionNumber).map { _ =>
-        Redirect(routes.ApplicationController.show(id))
-      }
-      case false => applications.saveSection(id, sectionNumber, fieldValues).map { _ =>
-        Redirect(routes.ApplicationController.show(id))
-      }
+    sectionTypeFor(sectionNumber) match {
+      case VanillaSection =>
+        allValuesEmpty(fieldValues) match {
+          case true => applications.deleteSection(id, sectionNumber).map { _ =>
```
Comment (kostya-lnk-ms) [inline src/main/scala/controllers/ActionHandler.scala]: This doesn't do any check for results of actions

File: src/main/scala/controllers/ActionHandler.scala [hunk 2]
```diff
@@ -5,35 +5,54 @@ import javax.inject.Inject
...
-      case errs => redisplaySectionForm(id, sectionNumber, JsonHelpers.flatten("", fieldValues), errs)
+      case CostSection =>
+        applications.getSection(id, sectionNumber).flatMap {
+          case Some(section) =>
+            applications.completeSection(id, sectionNumber, section.answers).flatMap {
+              case Nil => Future.successful(redirectToOverview(id))
+              case errs => redisplaySectionForm(id, sectionNumber, JsonHelpers.flatten("", section.answers), errs)
+            }
+
+          case None => Future.successful(NotFound)
```
Comment (kostya-lnk-ms) [inline src/main/scala/controllers/ActionHandler.scala]: Why would we throw "Page not found" at the user here? Should we have an error handling page instead?

File: src/main/scala/controllers/ActionHandler.scala (hunk 1 above)
Comment (DougC) [inline src/main/scala/controllers/ActionHandler.scala]: `saveSection` and `deleteSection` return `Future[Unit]` so there's currently nothing we can check errors on. The happy path is to return to the overview page. The Future will carry exceptions up on the sad path, and at some point we'll be bolstering the http filter to report them more cleanly than it currently does.

File: src/main/scala/controllers/ActionHandler.scala (hunk 2 above)
Comment (DougC) [inline src/main/scala/controllers/ActionHandler.scala]: Yes. I believe Play allows default pages for various responses to be configured so we can do that at a global level. We'll need some input from the UR and UX people as to how that should look.

**Pull request merged**